cabal-install/src/Distribution/Client/Setup.hs

```diff
@@ -604,12 +608,15 @@ data ConfigExFlags = ConfigExFlags {
     configAllowNewer    :: Maybe AllowNewer,
     configAllowOlder    :: Maybe AllowOlder,
     configWriteGhcEnvironmentFilesPolicy
-      :: Flag WriteGhcEnvironmentFilesPolicy
+      :: Flag WriteGhcEnvironmentFilesPolicy,
+    configPickFirstTarget
```

Comment:
I am a bit lost about the purpose of this flag... If something is ambiguous, just pick any possible target?

Seems like this is something that does not necessarily belong to this PR.

Reply:
Extracted the commits and will open a new PR for it